lib/cloudant.js

```diff
@@ -493,14 +500,14 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   };
 
   if (id) {
-    self.updateAttributes(model, id, data, {}, function(err, docs) {
+    self.update(model, {id: id}, data, {}, function(err, docs) {
       if (err && err.statusCode !== 404) return cb(err);
       else if (err && err.statusCode === 404) {
         // strip out the `_rev` property before creation
         delete data._rev;
         self.create(model, data, {}, createHandler);
       } else {
-        return cb(err, self.fromDB(model, mo, docs), {isNewInstance: false});
+        return cb(err, docs[0], {isNewInstance: false});
```

Comment:
sorry I was not involved in the whole discussion of pr https://github.com/strongloop/loopback-connector-cloudant/pull/123

> by stripping off `_rev`

Then cloudant will just insert the data right? It won't throw error for a data without `_id` and `_rev`